File: rev-server.js
#!/usr/bin/env node
'use strict';

var express = require('express');
var path = require('path');
var winston = require('winston');

var pkg = require('./package.json');
var cloader = require('./utils/configloader');
var mloader = require('./utils/moduleloader');

exports.config = null;
exports.log = null;
exports.express_app = null;
exports.models = null;
exports.RevModel = require('./core/rev_model').RevModel;

var config = exports.config = cloader.loadConfig();
//console.dir(config);

var log = exports.log = new (winston.Logger)({
	transports: [
	    new (winston.transports.Console)({
	    	timestamp: true,
	    	colorize: true
	    })
	    //new (winston.transports.File)({ filename: 'rev.log' })
    ]
});

log.info('RevFramework v%s Initialising...', pkg.version);
log.info('Config file: %s', config.config);

var models = exports.models = mloader.loadModules();

var express_app = exports.express_app = express();

if (config.log_requests) {
	express_app.use(function(req, res, next) {
	    log.info('[%s] %s %s', req.ip, req.method, req.originalUrl);
	    next();
	});
}

express_app.get('/', function(req, res){
	res.send('Hello World!');
});

express_app.use('/static/', express.static(path.join(__dirname, 'static')));

var server = express_app.listen(config.server_port, config.server_address, function() {
	log.info('Listening on %s port %s.', config.server_address, config.server_port);
});

File: revserver.py
#!/usr/bin/env python3

# Configure Base Import Path
import os,sys
basedir = os.path.dirname(__file__)
sys.path.insert(0, basedir)

# Configure Logging
import logging
from colorlog import ColoredFormatter

console_log = logging.StreamHandler()
console_log.setFormatter(ColoredFormatter(
    "%(asctime)s [%(log_color)s%(levelname)s%(reset)s] %(message)s",
))

logging.basicConfig(level=logging.DEBUG,
        format='%(asctime)s [%(levelname)s] %(message)s',
        handlers=[#log.FileHandler("revserver.log"),
            console_log])

import rev
rev.log = logging

# Initialise Application
from rev.core import PKG_NAME, PKG_VERSION
from rev.utils import configloader, moduleloader

rev.log.info('{} v{} Initialising...'.format(PKG_NAME, PKG_VERSION))

rev.config = configloader.load_config()

# TODO: Connect to database
rev.log.info("Database Server: {}:{}".format(rev.config['db_host'], rev.config['db_port']))
rev.log.info("Database Name: {}".format(rev.config['db_name']))

from pymongo import MongoClient
dbclient = MongoClient(rev.config['db_host'], rev.config['db_port'])
db = dbclient[rev.config['db_name']]

rev.registry = moduleloader.load_modules(db)